Frameworks/Foundation/NSMutableArray.mm

```diff
@@ -154,49 +155,53 @@ - (void)exchangeObjectAtIndex:(NSUInteger)atIndex withObjectAtIndex:(NSUInteger)
     [obj2 release];
 }
 
-/**
- @Status Interoperable
-*/
-- (void)removeObject:(NSObject*)objAddr {
-    if (objAddr == nil) {
-        TraceVerbose(TAG, L"objAddr = nil!");
-    }
+using _NSObjectComparator = bool (*)(id, id);
 
-    int idx = [self indexOfObject:objAddr];
-    if (idx != NSNotFound) {
-        [self removeObjectAtIndex:idx];
+- (void)_removeObject:(NSObject*)object inRange:(NSRange)range withComparator:(_NSObjectComparator)comparator {
+    NS_COLLECTION_VALIDATE_RANGE(range, self.count);
+    for (NSUInteger i = range.location + range.length; i > range.location; --i) {
+        id cur = [self objectAtIndex:i - 1];
+        if (comparator(object, cur)) {
+            [self removeObjectAtIndex:i - 1];
+        }
     }
 }
 
 /**
  @Status Interoperable
 */
-- (void)removeObject:(NSObject*)objAddr inRange:(NSRange)range {
-    for (int i = range.location + range.length - 1; i >= (int)range.location; i--) {
-        id curObj = [self objectAtIndex:i];
+- (void)removeObject:(NSObject*)object {
+    [self removeObject:object inRange:NSRange{ 0, self.count }];
+}
+/**
+ @Status Interoperable
+*/
+- (void)removeObject:(NSObject*)object inRange:(NSRange)range {
+    [self _removeObject:object inRange:range withComparator:[](id left, id right) -> bool { return [right isEqual:left]; }];
+}
 
-        if ([curObj isEqual:objAddr]) {
-            [self removeObject:curObj];
-        }
-    }
+/**
+ @Status Interoperable
+*/
+- (void)removeObjectIdenticalTo:(NSObject*)object {
+    [self removeObjectIdenticalTo:object inRange:NSRange{ 0, self.count }];
 }
 
 /**
  @Status Interoperable
 */
-- (void)removeObjectsInRange:(NSRange)range {
-    for (int i = range.location + range.length - 1; i >= (int)range.location; i--) {
-        [self removeObjectAtIndex:i];
-    }
+- (void)removeObjectIdenticalTo:(id)object inRange:(NSRange)range {
+    [self _removeObject:object inRange:range withComparator:[](id left, id right) -> bool { return right == left; }];
```

Comment:
Alas, yes. `[x isEqual:y]` returns BOOL. Functions that return one/the other aren't implicitly convertible.